### PR TITLE
Make it easy to record/update api-changes xml file with jbang script

### PR DESCRIPTION
**Pull request by zedbeit**
Added a JBang script to make it easy to update the api-changes.xml file.

When you run revapi, it generates api-changes-suggestions.xml file in the target directory. This file contains suggestions of the new changes you might want to ignore. 

So when you run jbang script, it will take the suggestions and put it in api-changes.xml file. The api-changes.xml file contains the new changes you want to ignore.

To run revapi use this command in any of the modules:
`mvn -Papi-check -Drevapi.skip=false -DskipTests verify`

To run the JBang script:
`jbang revapi-update`

### CODE REVIEW

File: script.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    public static void main(String[] args) {
+
+        Path currDir = Path.of(System.getProperty("user.dir"));
+
+        File apiChangesFile = currDir.resolve("api-changes.xml").toFile();
+        File apiSuggestionsFile = currDir.resolve("target/api-changes-suggestions.xml").toFile();
+
+
+        if(!apiSuggestionsFile.exists()){
+            throw new IllegalStateException("api-changes-suggestions.xml file does not exist. Run revapi first.");
```
Comment (metlos) [inline script.java:39]: Maybe would be even better to say how to "run revapi" here.

File: script.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
+//usr/bin/env jbang "$0" "$@" ; exit $?
```
Comment (metlos) [inline script.java:1]: `script.java` is a little bit non-descriptive. Something like `apply-revapi-suggestions.java` would IMHO be better so that people don't have to guess at the purpose of this script.

**Review by metlos — COMMENTED** [review]
Thanks @zedbeit! In general I think this does work. Just a couple of minor suggestions below.

Also, make sure the formatting of the java file conforms to Quarkus conventions

**Comment by zedbeit** [comment]
@metlos made some changes concerning the suggestions.

File: independent-projects/revapi/src/main/resources/revapi/revapi-configuration.xml [hunk 3]
```diff
@@ -3,7 +3,7 @@
 <analysisConfiguration>
   <revapi.java>
     <missing-classes>
-      <behavior>error</behavior>
```
Comment (maxandersen) [inline independent-projects/revapi/src/main/resources/revapi/revapi-configuration.xml:6]: why this change ?

File: jbang-catalog.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (maxandersen) [inline jbang-catalog.json:1]: I suggest you put this in `.jbang/jbang-catalog.json`

File: apply-revapi-suggestions.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
+//usr/bin/env jbang "$0" "$@" ; exit $?
```
Comment (maxandersen) [inline apply-revapi-suggestions.java:1]: I suggest you put this in `.jbang/RevapiUpdate.java` to make it easier to edit/use in IDE's directly.

File: apply-revapi-suggestions.java [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+/**
+ * This script applies the suggestions generated by the revapi output into the api-changes.xml file.
+ * The suggestions are in api-changes-suggestions.xml file. These suggestions are the new changes in the codebase that
+ * you might want to ignore (or otherwise process).
+ *
+ * The api-changes.xml file contains the new changes you want to ignore.
+ *
+ * This script record/update the suggestions that are in the api-changes-suggestions.xml into api-changes.xml.
+ */
+class jbangscript {
```
Comment (maxandersen) [inline apply-revapi-suggestions.java:35]: name it the same as file - yes, its valid but many will get confused and not focus on the real value.

File: independent-projects/revapi/src/main/resources/revapi/revapi-configuration.xml (hunk 3 above)
Comment (zedbeit) [reply independent-projects/revapi/src/main/resources/revapi/revapi-configuration.xml:6]: So that it should report missing classes not fail the build

File: build-parent/pom.xml [hunk 7]
```diff
@@ -155,6 +155,8 @@
         <revapi.skip>true</revapi.skip>
         <!-- By default don't check dependencies -->
         <revapi.checkdeps>false</revapi.checkdeps>
+        <!-- The message that is appended to the end of the plugin error output when the API problems fail the build -->
+        <revapi.buildFailureMessage>Run "jbang revapi-update" to update api-changes.xml file</revapi.buildFailureMessage>
```
Comment (metlos) [inline build-parent/pom.xml:159]: I would also suggest to the user that they need to add the justification for those changes in that file.

**Review by metlos — COMMENTED** [review]
From my POV, this looks good and does what we discussed. I'll let @maxandersen have the final word though :)

**Review by metlos — APPROVED** [review]

**Comment by maxandersen** [comment]
merged after squashing on command line. thanks @zedbeit
